lib/OpenQA/Worker/Settings.pm

```diff
@@ -86,7 +88,18 @@ sub new ($class, $instance_number = undef, $cli_options = {}) {
 
 sub auto_detect_worker_address ($self, $fallback = undef) {
     my $global_settings = $self->global_settings;
-    return 1 if defined $global_settings->{WORKER_HOSTNAME} && !$self->{_worker_address_auto_detected};
+    my $current_address = $global_settings->{WORKER_HOSTNAME};
+
+    # allow overriding WORKER_HOSTNAME explicitly; no validation is done in this case
+    return 1 if defined $current_address && !$self->{_worker_address_auto_detected};
+
+    # assign "localhost" as WORKER_HOSTNAME so entirely local setups work out of the box
+    if ($self->is_local_worker) {
+        $global_settings->{WORKER_HOSTNAME} = 'localhost';
```

Comment:
How about only defining this when no value is specified in WORKER_HOSTNAME. Like, combine with line 91

Reply:
I have no idea how that would make sense.